packages/winpanda/extra/src/winpanda/core/utils.py

```diff
@@ -17,6 +17,43 @@
 LOG = logger.get_logger(__name__)
 
 
+def exception_handler(func):
+    def wrapper(fpath, emheading=None, render=False, context=None):
+        assert isinstance(fpath, Path) and fpath.is_absolute(), (
+            f'Argument: fpath: Absolute pathlib.Path is required: {fpath}'
+        )
+        if context is not None:
+            assert isinstance(context, ResourceContext), (
+                f'Argument: context:'
+                f' Got {type(context).__name__} instead of ResourceContext'
+            )
+
+        try:
+            content = func(fpath, emheading, render, context)
+
+        except (FileNotFoundError, jj2.TemplateNotFound) as e:
+            err_msg = f'Load: {fpath}'
+            err_msg = f'{emheading}: {err_msg}' if emheading else err_msg
+            raise cr_exc.RCNotFoundError(err_msg) from e
+        except (OSError, RuntimeError) as e:
+            err_msg = f'Load: {fpath}: {type(e).__name__}: {e}'
+            err_msg = f'{emheading}: {err_msg}' if emheading else err_msg
+            raise cr_exc.RCError(err_msg) from e
+        except (jj2.TemplateError,) + cr_exc.JSON_ERRORS as e:
+            err_msg = f'Load: {fpath}: {type(e).__name__}: {e}'
+            err_msg = f'{emheading}: {err_msg}' if emheading else err_msg
+            raise cr_exc.RCInvalidError(err_msg) from e
+        except (jj2.TemplateError, yaml.YAMLError) as e:
+            err_msg = f'Load: {fpath}: {type(e).__name__}: {e}'
+            err_msg = f'{emheading}: {err_msg}' if emheading else err_msg
+            raise cr_exc.RCInvalidError(err_msg) from e
```

Comment:
These can be combined into a single `(jj2.TemplateError, yaml.YAMLError) + cr_exc.JSON_ERRORS` clause. The bodies are the same and the second mention of `jj2.TemplateError` has no effect.